backend/routes/friends.js

```diff
@@ -9,7 +9,12 @@ router.use(cors())
  * find user based on username search and filter out people you're already friends with
  */
 router.get('/users', async (req, res) => {
+    //send notification option if friend is not there
     try {
+        let userExistsCheck = await new Parse.Query(Parse.User).equalTo("username", req.query.name).first()
+        if (!userExistsCheck) {
+            res.status(200).send({})
+        }
```

Comment:
general note on this fn. Make the error catching as 'tight' as possible. As i mentioned before, if you are catching an exception, you should know why and in what cases that particular code would throw an error. If you are catching an error, for code A then solely wrap code A in the try block. In this case you are doing a whole lot, and catching with a generic error message below. This can quickly lead to unreliable systems with no easy way to debug or trace through where the error happened. It also can lead to sloppy code where you catch every code with a generic exception. Being specific about where you put in catches, and describing the possible error is much more helpful in increasing the reliability of your code in production.